### PR TITLE
CICD for new gitflow

### PR DESCRIPTION
**Pull request by tarcisiozf**
On pull request:
* check branch name conventions
* run lint and tests
Example: https://github.com/tarcisiozf/bfx-hf-algo/runs/5599873838?check_suite_focus=true

On merge to develop:
* create prelease
* push to the canary tag
* publish version and tags
Example: https://github.com/tarcisiozf/bfx-hf-algo/runs/5757937539?check_suite_focus=true

On merge to master:
* check for unstable dependencies
* bump minor version
* publish version and tags
Example: https://github.com/tarcisiozf/bfx-hf-algo/runs/5761262042?check_suite_focus=true

During development:
* add the "canary" script that installs all deps using the canary tag

Security:
* forked third-party action components
* [CODEOWNERS](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-codeowners-to-monitor-changes) for the `.github` folder

### CODE REVIEW

File: .github/workflows/develop.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.GIT_PERSONAL_ACCESS_TOKEN }}
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '12'
+          registry-url: 'https://registry.npmjs.org'
+      - run: git config --global user.name 'Tarcisio Ferraz'
```
Comment (tarcisiozf) [inline .github/workflows/develop.yml:18]: to be defined

File: .github/workflows/develop.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      - run: git config --global user.email 'tarcisiozf@users.noreply.github.com'
```
Comment (tarcisiozf) [inline .github/workflows/develop.yml:19]: to be defined

File: .github/workflows/master.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.GIT_PERSONAL_ACCESS_TOKEN }}
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '12'
+          registry-url: 'https://registry.npmjs.org'
+      - run: git config --global user.name 'Tarcisio Ferraz'
```
Comment (tarcisiozf) [inline .github/workflows/master.yml:18]: to be defined

File: .github/workflows/master.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      - run: git config --global user.email 'tarcisiozf@users.noreply.github.com'
```
Comment (tarcisiozf) [inline .github/workflows/master.yml:19]: to be defined

File: package.json [hunk 5]
```diff
@@ -60,6 +61,7 @@
     "bfx-hf-models": "git+https://github.com/bitfinexcom/bfx-hf-models.git#v2.4.0",
     "bfx-hf-models-adapter-lowdb": "git+https://github.com/bitfinexcom/bfx-hf-models-adapter-lowdb.git#v1.0.5",
     "chai": "^4.2.0",
+    "check-unstable-deps": "git+https://github.com/tarcisiozf/check-unstable-deps.git",
```
Comment (tarcisiozf) [inline package.json:64]: wait for repo transfer to org https://github.com/tarcisiozf/check-unstable-deps

File: package.json [hunk 6]
```diff
@@ -59,6 +60,7 @@
     "bfx-hf-ext-plugin-bitfinex": "github:bitfinexcom/bfx-hf-ext-plugin-bitfinex#v1.0.11",
     "bfx-hf-models": "git+https://github.com/bitfinexcom/bfx-hf-models.git#v2.4.0",
     "bfx-hf-models-adapter-lowdb": "git+https://github.com/bitfinexcom/bfx-hf-models-adapter-lowdb.git#v1.0.5",
+    "bfx-hf-version-utils": "github:tarcisiozf/bfx-hf-version-utils#move-to-repo",
```
Comment (tarcisiozf) [inline package.json:63]: wait for merge https://github.com/bitfinexcom/bfx-hf-version-utils/pull/1

File: .github/workflows/pull_request.yml [hunk 7]
```diff
@@ -0,0 +1,18 @@
+name: Pre-Merge
+on: [pull_request]
+jobs:
+  lint_and_tests:
+    name: check lint and tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: tarcisiozf/action-branch-name@master
+        with:
+          allowed_prefixes: 'feature,hotfix'
```
Comment (vitaliystoliarovcc) [inline .github/workflows/pull_request.yml:10]: shouldn't `docs`, `refactor`, `test` be added to the prefixes?

**Review by vitaliystoliarovcc — APPROVED** [review]

**Review by avsek477 — APPROVED** [review]
lgtm

**Pull request closed without merging**